airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -224,7 +226,26 @@ def ti_update_state(
             next_kwargs=ti_patch_payload.trigger_kwargs,
             trigger_timeout=timeout,
         )
+    elif isinstance(ti_patch_payload, TIRescheduleStatePayload):
+        task_instance = session.get(TI, ti_id_str)
+        actual_start_date = timezone.utcnow()
+        # add changes to TaskReschedule table to the session
+        session.add(
+            TaskReschedule(
+                task_instance.task_id,
+                task_instance.dag_id,
+                task_instance.run_id,
+                task_instance.try_number,
+                actual_start_date,
+                task_instance.end_date,
```

Comment:
Hmm? How would be `end_date`?

Task was set to running in `run` endpoint, it was then running via Supervisor/Task Runner, now it caught `AirflowRescheduleException`, so it will now set its state to `UP_FOR_RESCHEDULE` along with all the other things on TI table  like `end_date, duration`.

You would need something similar to 
https://github.com/apache/airflow/blob/38201cf1cfa23a6a363c31f66c39529a2b9b7ae3/airflow/api_fastapi/execution_api/routes/task_instances.py#L205

Reply:
You would need 

https://github.com/apache/airflow/blob/8848d6e35f6ead2cc1ed9887af8360fbd61c78c3/airflow/models/taskinstance.py#L1621-L1630

Reply:
Thanks for the point. I agree with you that we need "end_date". I have changed the code now to handle it like it is handled for the `TITerminalStatePayload` related states.

As for adding query like `query = TI.duration_expression_update(ti_patch_payload.end_date, query, session.bind) `, we do not need that to calculate the duration because the model itself has logic to do that as long as rthe start and end dates are set properly. I have adding a unit test that can validate this part as well. https://github.com/apache/airflow/blob/main/airflow/models/taskreschedule.py#L95-L98

Reply:
https://github.com/apache/airflow/pull/44907/commits/80f9ddc59d60ea1f6c6e2ee260845d0b4736d5b2 handles these scenarios. I also realised that there was no UT on the task_runner side. I added that as well.

Reply:
>we do not need that to calculate the duration because the model itself has logic to do that as long as rthe start and end dates are set properly. I have adding a unit test that can validate this part as well. https://github.com/apache/airflow/blob/main/airflow/models/taskreschedule.py#L95-L98

That is TaskReschedule, but we need duration for TI

Reply:
Just ran an example and realised that. I will fix it.